src/boolean_result.cpp

```diff
@@ -809,6 +810,40 @@ Manifold::Impl Boolean3::Result(OpType op) const {
   PRINT(n12 << " new verts from edgesP -> facesQ");
   PRINT(n21 << " new verts from facesP -> edgesQ");
 
+  const int numProp = std::max(inP_.NumProp(), inQ_.NumProp());
+  if (numProp > 0) {
+    // Ideally, switch over from triProperties vec to Halfedge.startProp - then
+    // the prop index will be available in the append edges functions below.
+    Vec<std::atomic<int>> prop2VertP(inP_.NumPropVert());
+    for_each_n(autoPolicy(inP_.NumTri()), countAt(0), inP_.NumTri(),
+               [&](size_t tri) {
+                 for (const int i : {0, 1, 2}) {
+                   prop2VertP[inP_.meshRelation_.triProperties[tri][i]].store(
+                       inP_.halfedge_[3 * tri + i].startVert);
+                 }
+               });
+    Vec<std::atomic<int>> prop2VertQ(inQ_.NumPropVert());
+    for_each_n(autoPolicy(inQ_.NumTri()), countAt(0), inQ_.NumTri(),
+               [&](size_t tri) {
+                 for (const int i : {0, 1, 2}) {
+                   prop2VertQ[inQ_.meshRelation_.triProperties[tri][i]].store(
+                       inQ_.halfedge_[3 * tri + i].startVert);
+                 }
+               });
+    // Scan the properties inclusion numbers, e.g. + i03[prop2VertP[prop]].
+    //
+    // New verts create either 2 or 3 new properties: one for the face, plus one
+    // for an edge where start and end verts share properties, or plus two for
+    // an edge that doesn't share both properties.
+    //
+    // Copy included inP and inQ properties to outR, and duplicate for new verts
+    // accoding to the inclusion numbers. Calculate new outR properties using
+    // the Barycentric function from CreateProperties.
```

Comment:
For 2-iii: It's best to think of propVerts as being on a halfedge or triangle - the paired halfedge usually, but doesn't always, shares the same propVerts. Only if both ends of the paired halfedges share propVerts can you interpolate that as a single new shared propVert - otherwise you'll be creating two new propVerts, one for each halfedge. It's not possible to have two ends of the same halfedge have properties that don't represent the same channels of data. The channels of data are consistent across each meshID, or equivalently, each triangle run in MeshGL. So they are also consistent over a triangle or halfedge.

Reply:
Can you define what share propVerts means? If each propVert maps to exactly one vertex.

And for these new vertices, where do their propVerts come from? Say, if each vertex of the associated triangle originally maps to 2 propVert, we should have 2 propVert interpolated using barycentric coord, plus some propVerts interpolated using propVerts associated to endpoints of the edge?

Reply:
By share propVerts, I mean the paired halfedges point to the same prop index. Basically verts *must* be manifold, while propVerts *may* be manifold. The edge is prop-manifold if the propVerts are shared.

The second part isn't quite right. Any given triangle only has one set of three propVerts; new propVerts on either its edges or its center all get interpolated barycentricly using that single interpolant. So e.g. 6 triangles ajoin a vertex - this vertex might have 2 propVerts, meaning e.g. 3 of those triangles point to one propVert and 3 point to the other.

Reply:
If two paired halfedges (with different starting vertices) can point to the same prop vert, what does it mean by each prop vert maps to one vert?

Or does this mean each prop vert maps to one edge (can be one or two halfedges)?